server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java

```diff
@@ -98,44 +105,91 @@ public class TableService {
     @Transactional(rollbackFor = Exception.class)
     @SkipAuthorize("permission check inside")
     public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLException, InterruptedException {
+        List<DBObjectType> types = params.getTypes();
```

Comment:
we should return both basic table and external table when `param.getTypes()` returns null or empty collection.

return basic table when `param.getTypes()` returns only basic Table
return external table when `param.getTypes()` returns only external table